windows/src/developer/TIKE/child/Keyman.Developer.UI.UframeWordlistEditor.pas

```diff
@@ -291,18 +294,17 @@ function TframeWordlistEditor.SaveToFile(const Filename: string): Boolean;
 
 procedure TframeWordlistEditor.SetModified(const Value: Boolean);
 begin
-  if FModified <> Value then
-  begin
-    FModified := Value;
-    if Assigned(FOnModifiedChanged) then
-      FOnModifiedChanged(Self);
-  end;
+  FModified := Value;
 end;
 
 procedure TframeWordlistEditor.SourceChanged(Sender: TObject);
 begin
   if FSetup = 0 then
+  begin
     Modified := True;
+    if Assigned(FOnChanged) then
+      FOnChanged(Self);
+  end;
 end;
```

Comment:
I'm guessing that this was the part being missed previously?

Reply:
So the wordlist editor is made up of two sub-editors: a code editor and a grid editor, and is itself a sub-editor of a bigger editor -- the model editor. This makes mapping the modification state machine a little daunting.

This cod echange ensures that every change is bubbled up through to the owner editor where it maintains a global 'modified' state. It was easier to just make sure that global 'modified' state was tracked on every change rather than on the change of the modified state within in the internal editor or one of its sub-editors.